scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GetSpecificationGoalsAction.java

```diff
@@ -33,21 +42,36 @@ public GetSpecificationGoalsAction(final Connection connection) throws SQLExcept
     this.statement = connection.prepareStatement(sql);
   }
 
-  public List<GoalRecord> get(final long specificationId) throws SQLException {
+  public List<GoalInvocationRecord> get(final long specificationId) throws SQLException {
     this.statement.setLong(1, specificationId);
     final var resultSet = this.statement.executeQuery();
 
-    final var goals = new ArrayList<GoalRecord>();
-    while (resultSet.next()) {
-      final var id = resultSet.getLong("goal_id");
-      final var revision = resultSet.getLong("revision");
-      final var name = resultSet.getString("name");
-      final var definition = resultSet.getString("definition");
-      final var simulateAfter = resultSet.getBoolean("simulate_after");
-      goals.add(new GoalRecord(new GoalId(id, revision), name, new GoalSource(definition), simulateAfter));
-    }
+    try {
 
-    return goals;
+      final var goals = new ArrayList<GoalInvocationRecord>();
+      while (resultSet.next()) {
+        final var id = resultSet.getLong("goal_id");
+        final var goalInvocationId = resultSet.getLong("goal_invocation_id");
+        final var revision = resultSet.getLong("revision");
+        final var name = resultSet.getString("name");
+        final var definition = resultSet.getString("definition");
+        final var simulateAfter = resultSet.getBoolean("simulate_after");
+        final var type = resultSet.getString("type");
+        final var path = resultSet.getString("path");
+        final var args = parseJson(resultSet.getString("arguments"), new SerializedValueJsonParser());
+
+        goals.add(new GoalInvocationRecord(
+            new GoalId(id, revision, goalInvocationId),
+            name,
+            type.equals("JAR") ? new GoalType.JAR(Path.of(path)) : new GoalType.EDSL(new GoalSource(definition)),
+            args.asMap().get(),
+            simulateAfter
+        ));
+      }
+      return goals;
+    } catch (InvalidJsonException | InvalidEntityException e) {
+      throw new RuntimeException(e);
```

Comment:
This should probably be `SQLException` or some other exception type instead of `Runtime`

Reply:
done